graylog2-server/src/main/java/org/graylog2/configuration/WebListenUriValidator.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.configuration;
+
+import com.github.joschi.jadconfig.ValidationException;
+import com.github.joschi.jadconfig.Validator;
+
+import java.net.URI;
+
+public class WebListenUriValidator implements Validator<URI> {
+    @Override
+    public void validate(String name, URI value) throws ValidationException {
+        if (value != null && !value.getPath().equals("/")) {
```

Comment:
Unfortunately [`URI#getPath()`](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getPath--) can return `null` which would lead to a `NullPointerException` here. Maybe just invert the second condition (`"/".equals(…)`).